### DISCUSSION TITLE
Fzf-lua keymaps: Detail column hard to see fully

### DISCUSSION THREAD

**Discussion by 1ngemar** [ibhagwan/fzf-lua#1566] [Q&A] [answered]
Hi,

When searching for your keymaps -  'fzflua keymaps' is great. But the detail column easily gets cut off.
(I haven't got any 'wrap' options to work (?))

Is is feasible to add an option 'show_description = true/false' - or wrap the line? 

Or an even better solution?

**Comment by ibhagwan** [comment]
Regarding wrap and screen real estate, this just came up: https://github.com/ibhagwan/fzf-lua/discussions/1565

> Is is feasible to add an option 'show_description = true/false' - or wrap the line?
> Or an even better solution?

I’ll think of something.

**Comment by ibhagwan** [comment, marked as answer]
I realize it might be the opposite of what you're asking, just FYI, since #1452 it's already possible to turn off the details column with `:FzfLua keymaps show_details=false`.

~~I can add `show_desc` if that's what you're after.~~

**EDIT:**
`show_desc` option added with https://github.com/ibhagwan/fzf-lua/commit/95f1d1dfa215b035e17ebd9e3a6bc32fec7c0fe3:
> Can also be conigured via setup under `keymaps.show_desc`
```lua
:FzfLua keymaps show_desc=false
```

**Reply by 1ngemar** [reply]
Yes; it's the complete details in this case I'm looking for.

Thanks.
/i

**Comment by 1ngemar** [comment]
And is it possible to have a command - to go to the file/line where the keymapping is set/defined? I'm thinking of your (own) custom key mappings, like eg:

![leader-b](https://github.com/user-attachments/assets/974ea96b-e672-4710-8328-07c59df6e656)

**Reply by ibhagwan** [reply]
> And is it possible to have a command - to go to the file/line where the keymapping is set/defined? I'm thinking of your (own) custom key mappings, like eg:
> 
> ![leader-b](https://private-user-images.githubusercontent.com/175594591/395808579-974ea96b-e672-4710-8328-07c59df6e656.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQxOTkwNzMsIm5iZiI6MTczNDE5ODc3MywicGF0aCI6Ii8xNzU1OTQ1OTEvMzk1ODA4NTc5LTk3NGVhOTZiLWU2NzItNDcxMC04MzI4LTA3YzU5ZGY2ZTY1Ni5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNFQxNzUyNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jN2JiNWE2MDg0OTMyYmE4YzlhYTMzMWViZGI2OWRiODM2NjdiZjMyOGVkNjJlNmFkZDdhODUwODQ2M2EwNWYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Zu4i06oDWsqMM4GacTPmMSxloImrgugBqMy2RQWo-qY)

This already happens for lua style key maps using a functions.

**Reply by 1ngemar** [reply]
Ok, I was thinking of a way to jump directly from the fzf-lua keymaps view directly to the definition of the keymapping. So this is possible?

(But although; just a nice-to-have function)

**Reply by ibhagwan** [reply]
> Ok, I was thinking of a way to jump directly from the fzf-lua keymaps view directly to the definition of the keymapping. So this is possible?
> 
> (But although; just a nice-to-have function)

As I mentioned, if the keymap is a lua function you can go to defeintion with `ctrl-s|v|t` (split, vsplit, tab).

Unforuntaely if the mapping is a string it's not possible as the `:verbose map <key>` doesn't nclude the source, for example:
![image](https://github.com/user-attachments/assets/43c9c841-5d48-4d96-9ca3-fbba5d06bde3)

**Comment by 1ngemar** [comment]
Aha.

Got it, thanks.

**Comment by ibhagwan** [comment]
@1ngemar, with https://github.com/ibhagwan/fzf-lua/commit/95f1d1dfa215b035e17ebd9e3a6bc32fec7c0fe3 you can now use:
```lua
:FzfLua keymaps show_desc=false
```

Can also be conigured via setup under `keymaps.show_desc`.

**Comment by 1ngemar** [comment]
Yes, thanks.

Already tried it. Works great.

Den sön 15 dec. 2024 kl 05:49 skrev ibhagwan ***@***.***>:

**Comment by 1ngemar** [comment]
Then,

regarding our earlier conversation around matching words in grep; fzf, rg.

Maybe this is an idea:
Make all chars lowercase, eg in hello or Hello or hEllo makes 'hello'

Then fabricate a string like so (for all chars a..z) :
'[Hh][Ee][Ll][Ll][Oo]'

Then, the fzf window will sort (out) the possible multiple regex matches,
say if you only search for 'hello'; it will still only show that matched
line for 'hello'. But if you use --smart-case and there are multiple
matches in the preview window; they will all be highlighted.

Maybe that would be a way forward? But I see that the regex line would look
messy :)

/i


Den sön 15 dec. 2024 kl 08:44 skrev Ingemar Åström ***@***.***

**Reply by ibhagwan** [reply]
> Maybe that would be a way forward? But I see that the regex line would look
messy :)

Messy, prone to error and perhaps also not the prefernace of some users, say you're not using smart case? then you'd also get matches that don't wish to see, that means we also have to adapt the regex to the rg options, IMHO this feels like overly complex for a minor inconvinience so I don't plan on embarking in this anytime soon.

**Comment by 1ngemar** [comment]
Yes.

All good.

/i

Den sön 15 dec. 2024 10:25ibhagwan ***@***.***> skrev:

**Comment by loyd** [comment]
@ibhagwan after https://github.com/ibhagwan/fzf-lua/commit/95f1d1dfa215b035e17ebd9e3a6bc32fec7c0fe3 the description is always limited that makes https://github.com/ibhagwan/fzf-lua/pull/1452 meaningless =(

![image](https://github.com/user-attachments/assets/44432116-076c-4360-859e-081d5124b06e)

**Reply by ibhagwan** [reply]
Ty @loyd, will be fixed in #1802.
